packages/core/saved-objects/core-saved-objects-migration-server-internal/src/document_migrator/validate_migrations.ts

```diff
@@ -64,42 +64,44 @@ export function validateTypeMigrations({
     const modelVersionMap =
       typeof type.modelVersions === 'function' ? type.modelVersions() : type.modelVersions ?? {};
 
-    if (Object.keys(modelVersionMap).length > 0 && !type.switchToModelVersionAt) {
-      throw new Error(
-        `Type ${type.name}: Uusing modelVersions requires to specify switchToModelVersionAt`
-      );
-    }
-
-    Object.entries(modelVersionMap).forEach(([version, definition]) => {
-      assertValidModelVersion(version);
-    });
+    if (Object.keys(modelVersionMap).length > 0) {
+      if (!type.switchToModelVersionAt) {
+        throw new Error(
+          `Type ${type.name}: Using modelVersions requires to specify switchToModelVersionAt`
+        );
+      }
```

Comment:
Fixed a bug causing the validation to throw if an empty model version map was defined

Reply:
This will throw for types that haven't switched to models **on the currently running version yet**.  Remind me how we'll handle that if a type doesn't have a migration when the switch happens. Will we not allow them to fall back to the most recent (non-model) migration? 
I'm concerned that we're forcing folks to add a model when we force the switch over, even if the switch is postposed.

Reply:
> This will throw for types that haven't switched to models on the currently running version yet. How do we handle that if a type doesn't need a migration when the switch happens?

No it won't. I was previously throwing when `modelVersions: {}` was defined, and it won't anymore, this is all that changed. 

You never were forced to define a model version when you switch. It's just that you can't register old migrations anymore

Reply:
> The object needs to run through the 8.8 migration but 8.8 < 8.10 (when the switch happened) and we don’t have a model version. How do we handle that?

As we were doing previously. The document will be upgrade to `8.8` given it's the latest register migration.

Once again, keep in mind that the document migration doesn't know or care about model version. when model versions are registered, internally this part of the code only knows that `10.x.0` migration are present, and apply the exact same logic as it was previously doing

Reply:
> it will upgrade the doc to 8.6 during import, and set the typeMigrationVersion accordingly to 8.6. So yes, this is already, and still will be, supported.

Gold to my eyes! Thank you!

Reply:
> it will upgrade the doc to 8.6 during import, and set the typeMigrationVersion accordingly to 8.6. So yes, this is already, and still will be, supported.

Gold to my eyes! Thank you!